yrs/src/block.rs

```diff
@@ -101,9 +250,29 @@ impl Item {
             }
         }
     }
+
     pub fn len (&self) -> u32 {
         self.content.len()
     }
+
+    pub fn split(&mut self, diff: u32) -> Item {
+        let client = self.id.client;
+        let clock = self.id.clock;
+        let other = Item {
+            id: ID::new(client, clock + diff),
+            left: Some(BlockPtr::from(self.id)),
```

Comment:
This should point to the end of `self` after the split. So left is `left: Some(BlockPtr::from(ID::new(client, clock + diff - 1)))`